### PR TITLE
feat(controlplane): mailer options

### PR DESCRIPTION
**Pull request by d4rkstar**
## Motivation and Context
Added support for all mailer options. Now is possibile to change smtp hostname, port, credentials, tls and secure.
I added because SMTP server is hardcoded and is not possible to change it via env.

## TODO

- [X] Tests or benchmark included
- [ ] Documentation is changed or added on [https://app.gitbook.com/](https://app.gitbook.com/)
- [X] PR title must follow [conventional-commit-standard](https://github.com/wundergraph/wundergraph/blob/main/CONTRIBUTING.md#conventional-commit-standard)

### CODE REVIEW

**Comment by PeteMac88** [comment]
@d4rkstar Can you maybe also update the helm charts to support the env variables? I created a similar PR but only included host and port (https://github.com/wundergraph/cosmo/pull/648)

**Comment by d4rkstar** [comment]
Yes I will!

**Comment by d4rkstar** [comment]
Hello @PeteMac88. Added other variables in helm charts. Let me know if there is other to adjust.

**Comment by PeteMac88** [comment]
@d4rkstar Any news when this change will be released?

**Comment by d4rkstar** [comment]
@PeteMac88 no :( However i'll resolve new conflicts in code and will wait again for approval

**Comment by Aenimus** [comment]
Hi,

We have assigned this to a member of the team, and it will be addressed in due course.

Thank you for your patience.

**Comment by StarpTech** [comment]
Hi @d4rkstar, sorry for the late review. I'm going to use your PR as foundation to finish it. Thank you.

**Comment by d4rkstar** [comment]
Hello @StarpTech, i'm happy to contribute!

File: helm/cosmo/charts/controlplane/templates/deployment.yaml [hunk 1]
```diff
@@ -236,6 +236,26 @@ spec:
...
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "controlplane.secretName" . }}
+                  key: smtpPort
+            - name: SMTP_HOST
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "controlplane.secretName" . }}
+                  key: smtpHost
+            - name: SMTP_SECURE
```
Comment (StarpTech) [inline helm/cosmo/charts/controlplane/templates/deployment.yaml:249]: Please set those only when it's true.

File: helm/cosmo/charts/controlplane/templates/deployment.yaml [hunk 2]
```diff
@@ -236,6 +236,26 @@ spec:
                 secretKeyRef:
                   name: {{ include "controlplane.secretName" . }}
                   key: smtpUsername
+            - name: SMTP_PORT
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "controlplane.secretName" . }}
```
Comment (StarpTech) [inline helm/cosmo/charts/controlplane/templates/deployment.yaml:242]: Non-secrets should come from the configMap

**Comment by StarpTech** [comment]
@d4rkstar awesome. Let me review it.

File: controlplane/src/core/build-server.ts [hunk 3]
```diff
@@ -212,8 +219,8 @@ export default async function build(opts: BuildConfig) {
   });
 
   let mailerClient: Mailer | undefined;
-  if (opts.smtpUsername && opts.smtpPassword) {
-    mailerClient = new Mailer({ username: opts.smtpUsername, password: opts.smtpPassword });
+  if (opts.mailer.smtpUsername && opts.mailer.smtpPassword) {
+    mailerClient = new Mailer(opts.mailer as MailerParams);
```
Comment (StarpTech) [inline controlplane/src/core/build-server.ts:223]: Can we avoid the cast? e.g by passing only the relevant properties.

**Comment by StarpTech** [comment]
HI @d4rkstar thank you for your contributions. You didn't respond to the review. We decided to base on your work here to finish it. See https://github.com/wundergraph/cosmo/pull/1008

**Comment by d4rkstar** [comment]
> HI @d4rkstar thank you for your contributions. You didn't respond to the review. We decided to base on your work here to finish it. See #1008

Sorry @StarpTech :( I had an unexpected workload and haven't been able to look at it yet. If you solved it with the code already submitted previously, that's great!

**Pull request closed without merging**